lldb/source/DataFormatters/VectorType.cpp

```diff
@@ -269,11 +269,15 @@ class VectorTypeSyntheticFrontEnd : public SyntheticChildrenFrontEnd {
     return lldb::ChildCacheState::eRefetch;
   }
 
-  size_t GetIndexOfChildWithName(ConstString name) override {
+  llvm::Expected<size_t> GetIndexOfChildWithName(ConstString name) override {
     const char *item_name = name.GetCString();
     uint32_t idx = ExtractIndexFromString(item_name);
-    if (idx < UINT32_MAX && idx >= CalculateNumChildrenIgnoringErrors())
-      return UINT32_MAX;
+    if (idx == UINT32_MAX ||
+        (idx < UINT32_MAX && idx >= CalculateNumChildrenIgnoringErrors()))
+      return llvm::createStringError(
+          "'SyntheticChildrenFrontEnd::VectorTypeSyntheticFrontEnd' cannot "
+          "find index of child '%s'",
```

Comment:
Can we also print the `idx` that we found in the error message? (same applies to other places that do the `idx >= CalculateNumChildrenIgnoringErrors` comparison)

Reply:
Type has no child named '%s'